### ISSUE TITLE
Async functions

### ISSUE THREAD

**Issue by scothis** [projectriff/node-function-invoker#7]
Just about any non-trivial function in Node will involve an asynchronous API call. Currently the node-function-invoker only supports synchronous function invocations, meaning no asynchronous methods may be used. There are two different models for handling asynchronicity which are not easily interoperable: callbacks and async/promises.

Callbacks are a classic Node pattern and in very wide spread use. Async methods are new in ES 2017 and well supported in Node 8. Async methods are syntactic sugar which boils down to a function that returns a Promise.

A callback method would look like:

```js
module.exports = (name, callback) => callback(null, `Hello ${name}!`);
```

A huge advantage of async functions is that the same invocation signature can also support sync methods transparently, whereas supporting callback functions and sync functions simultaneously is awkward and fragile.

```js
// sync
module.exports = name => `Hello ${name}!`;

// promise
module.exports = name => Promise.resolve(`Hello ${name}!`);

// async
module.exports = async name => `Hello ${name}!`;
```

The current invoker code is:

```js
var resultx = fn(req.body);
// send response
```

Callbacks would look like:

```js
fn(req.body, (err, resultx) => {
  // send response
});
```

Sync/async/promised functions would look like:

```js
var resultx = await fn(req.body);
// send response
```

I'd lean towards not supporting callbacks by default, and only add support as an opt-in if there is strong demand.

**Comment by jldec** [comment]
@scothis, you're right - any function which internally needs to wait for some async request (e.g. to make a service call on the network) would have no way to return and unblock the event loop while it waits.

**Comment by jldec** [comment]
Fixed with #9.
